olp-cpp-sdk-core/src/http/winhttp/NetworkWinHttp.h

```diff
@@ -110,21 +111,33 @@ class NetworkWinHttp : public Network {
     bool ignore_data;
     bool no_compression;
     bool uncompress;
+    bool in_use;
 #ifdef NETWORK_HAS_ZLIB
     z_stream strm;
 #endif
   };
 
+  RequestData* AllocateRequest(RequestId id,
+                          std::shared_ptr<ConnectionData> connection,
+                          Network::Callback callback,
+                          Network::HeaderCallback header_callback,
+                          Network::DataCallback data_callback,
+                          std::shared_ptr<std::ostream> payload,
+                          const NetworkRequest& request);
+  void FreeRequest(RequestData* handle );
+  void FreeRequest(RequestId id );
+  RequestData* FindRequest(RequestId id);
+
   static void CALLBACK RequestCallback(HINTERNET, DWORD_PTR, DWORD, LPVOID,
                                        DWORD);
   static DWORD WINAPI Run(LPVOID arg);
 
   void CompletionThread();
-
+  
   std::recursive_mutex mutex_;
   std::unordered_map<std::wstring, std::shared_ptr<ConnectionData>>
       http_connections_;
-  std::unordered_map<RequestId, std::shared_ptr<RequestData>> http_requests_;
+  std::vector<RequestData> http_requests_;
```

Comment:
It looks very strange to pass around pointers to vector data.
Why did you not keep the shared_ptr approach?